compiler/src/dotty/tools/backend/jvm/DottyBackendInterface.scala

```diff
@@ -840,7 +840,8 @@ class DottyBackendInterface(outputDirectory: AbstractFile, val superCallsMap: Ma
     def addRemoteRemoteExceptionAnnotation: Unit = ()
 
     def samMethod(): Symbol = ctx.atPhase(ctx.erasurePhase) {
-      toDenot(sym).info.abstractTermMembers.toList match {
+      val samMethods = toDenot(sym).info.abstractTermMembers.filter(! _.show.contains("equals"))
```

Comment:
I haven't tried it, but to filter all symbols that override a java.lang.Object symbol, you probably want:
```suggestion
      val samMethods = toDenot(sym).info.abstractTermMembers
        .filterNot(_.symbol.overriddenSymbol(defn.ObjectClass).exists)
```